lib/experimental/class-theme-json-webfonts-handler.php

```diff
@@ -0,0 +1,537 @@
+	 * Runs the theme.json webfonts handler to expose defined webfonts
+	 * to the editor's typography pickers and generate `@font-face` style
+	 * declarations.
+	 *
+	 * This is not a public API, but rather an internal handler.
+	 * A future public Webfonts API will replace this code.
+	 *
+	 * This code design is intentional to encapsulate the inner-workings
+	 * of this handler to ensure it is not accessible from outside of this
+	 * function. Why? This is a stopgap implementation to provide the mechanisms
+	 * for webfonts to be defined in a theme's `theme.json` file for style
+	 * variations.
+	 *
+	 * @link https://github.com/WordPress/gutenberg/issues/40472
+	 *
+	 * @since 6.0.0
+	 */
+	function _wp_theme_json_webfonts_handler() {
+		static $handler = null;
+
+		// This check prevents the class from being loaded into memory more than once.
+		if ( null !== $handler ) {
+			return;
+		}
+
+		/**
+		 * Handler for `theme.json` webfonts.
+		 *
+		 * This class is intentionally placed within the function to create
+		 * a private, non-public, non-accessible container. This code will be
+		 * removed when the Webfonts API is merged into Core.
+		 *
+		 * @since 6.0.0
+		 */
+		class Theme_JSON_Webfonts_Handler {
+
+			/**
+			 * Webfonts to be processed.
+			 *
+			 * @since 6.0.0
+			 *
+			 * @var array[]
+			 */
+			private $webfonts = array();
+
+			/**
+			 * Stylesheet handle.
+			 *
+			 * @since 6.0.0
+			 *
+			 * @var string
+			 */
+			private $stylesheet_handle = '';
+
+			/**
+			 * Initialize the handler.
+			 *
+			 * @since 6.0.0
+			 */
+			public function init() {
+				add_action( 'wp_loaded', array( $this, 'register_webfonts' ) );
+
+				// Register callback to generate and enqueue styles.
+				if ( did_action( 'wp_enqueue_scripts' ) ) {
+					$this->stylesheet_handle = 'wp-webfonts-footer';
+					$hook                    = 'wp_print_footer_scripts';
+				} else {
+					$this->stylesheet_handle = 'wp-webfonts';
+					$hook                    = 'wp_enqueue_scripts';
+				}
+				add_action( $hook, array( $this, 'generate_and_enqueue_styles' ) );
+				add_action( 'admin_init', array( $this, 'generate_and_enqueue_editor_styles' ) );
+
+				add_filter( 'mime_types', array( $this, 'add_mime_types' ) );
+			}
+
+			/**
+			 * Register webfonts within the `theme.json`.
+			 *
+			 * @since 6.0.0
+			 */
+			public function register_webfonts() {
+				$this->webfonts = array();
+				$webfonts       = $this->get_webfonts_from_theme_json();
+
+				foreach ( $webfonts as $webfont ) {
+					if ( ! is_array( $webfont ) ) {
+						continue;
+					}
+
+					$webfont = $this->convert_keys_to_kebab_case( $webfont );
+
+					$webfont = $this->validate_webfont( $webfont );
+
+					$webfont['src'] = $this->transform_src_into_uri( (array) $webfont['src'] );
+
+					// Skip if not valid.
+					if ( empty( $webfont ) ) {
+						continue;
+					}
+
+					$this->webfonts[] = $webfont;
+				}
+			}
+
+			/**
+			 * Gets the webfonts from `theme.json`.
```

Comment:
```suggestion
			 * Gets the webfonts from theme.json.
```
- Officially, markdown syntax is only valid in the body of a docblock, not the title.
- Other docblock titles in the PR do not encapsulate theme.json at all.